File: base64_and_file/src/main/java/com/phone/base64_and_file/model/IBase64AndFileModel.java
package com.phone.base64_and_file.model;

import android.content.Context;

import com.phone.base64_and_file.bean.Base64AndFileBean;
import com.phone.common_library.callback.OnCommonSingleParamCallback;

public interface IBase64AndFileModel {

    void showCompressedPicture(Context context,
                               Base64AndFileBean base64AndFileBean,
                               OnCommonSingleParamCallback<Base64AndFileBean> onCommonSingleParamCallback);

    void showPictureToBase64(Base64AndFileBean base64AndFileBean,
                             OnCommonSingleParamCallback<Base64AndFileBean> onCommonSingleParamCallback);

    void showBase64ToPicture(Base64AndFileBean base64AndFileBean,
                             OnCommonSingleParamCallback<Base64AndFileBean> onCommonSingleParamCallback);

}

File: base64_and_file/src/main/java/com/phone/base64_and_file/model/IBase64AndFileModel.kt
package com.phone.base64_and_file.model

import android.content.Context
import com.phone.base64_and_file.bean.Base64AndFileBean
import com.phone.library_common.callback.OnCommonSingleParamCallback

interface IBase64AndFileModel {

    fun showCompressedPicture(
        context: Context,
        base64AndFileBean: Base64AndFileBean,
        onCommonSingleParamCallback: OnCommonSingleParamCallback<Base64AndFileBean>
    )

    fun showPictureToBase64(
        base64AndFileBean: Base64AndFileBean,
        onCommonSingleParamCallback: OnCommonSingleParamCallback<Base64AndFileBean>
    )

    fun showBase64ToPicture(
        base64AndFileBean: Base64AndFileBean,
        onCommonSingleParamCallback: OnCommonSingleParamCallback<Base64AndFileBean>
    )

    fun detachViewModel()
}